usrsctplib/netinet/sctp_input.c

```diff
@@ -2821,7 +2821,7 @@ sctp_handle_cookie_echo(struct mbuf *m, int iphlen, int offset,
 		if ((uint32_t)diff.tv_sec > UINT32_MAX / 1000000) {
 			staleness = UINT32_MAX;
 		} else {
-			staleness = diff.tv_sec * 1000000;
+			staleness = (uint32_t)diff.tv_sec * 1000000;
```

Comment:
Fixed in https://github.com/sctplab/usrsctp/commit/87859a58c561d5a235c0c1b16036f876bde23fb1.